lib/record_store/zone.rb

```diff
@@ -56,22 +56,49 @@ def yaml_files
     validate :validate_provider_can_handle_zone_records
 
     def self.from_yaml_definition(name, definition)
+      definition['records'] ||= []
+      Dir["#{RecordStore.zones_path}/#{name}/*__*.yml"].each do |record_file|
+        fqdn, type = File.basename(record_file, '.yml').split('__')
```

Comment:
Feel free to ignore if you think I'm bikeshedding, but I'd prefer for the format to be `RR_TYPE__LABEL.yml` (e.g. `A__a-record.dnsimple.example.com.yml`.

None 🚲 🏠 : since we have the current origin in the directory name, we can shorten the name to omit the origin. Example: an A record for `test.example.com` could be `example.com/A__test.yml`.